src/network/networkd-ndisc.c

```diff
@@ -451,13 +451,18 @@ static int ndisc_router_process_onlink_prefix(Link *link, sd_ndisc_router *rt) {
         if (r < 0)
                 return log_link_warning_errno(link, r, "Failed to get prefix length: %m");
 
+        r = sd_ndisc_router_get_preference(rt, &preference);
```

Comment:
Are there any relevant statement in RFC about that the 'main' preference should/may be used for routes derived by the on-link prefix ?

Reply:
I did not find any. If omitted then pref always be  medium even if it's from high/low pref. Hence it's not behaving as is supposed to do. https://github.com/systemd/systemd/issues/28439 and https://github.com/systemd/systemd/issues/28426. 

Here is a verofied test https://github.com/systemd/systemd/issues/28426#issuecomment-1653398818